src/lib/progress-bar/progress-bar.ts

```diff
@@ -79,6 +87,19 @@ export class MatProgressBar {
       return {transform: `scaleX(${scale})`};
     }
   }
+
+  /**
+   * Reappends the already attached circle SVG element in DOM to update the color styling.
+   *
+   * For colors to take effect in the background div, the svg's <circle> element, must be removed
+   * then placed back into the DOM again.  This is necessary as the browser only defines the
+   * styling of SVG pattern elements as it is placed into the DOM.
+   */
+  private _reappendCircleSvgElement() {
```

Comment:
I'm not convinced that this is necessary. The progress spinner supports changing the colors without having to re-append the element to the DOM. Are you sure that it doesn't have something to do with the circles not having a unique id?

Reply:
I think you are right